models/ride.go

```diff
@@ -0,0 +1,189 @@
+		PricePerSeat int       `json:"price_per_seat" db:"price_per_seat"`
+		Info         string    `json:"info" db:"info"`
+		StartDate    time.Time `json:"start_date" db:"start_date"`
+		CreatedAt    time.Time `json:"created_at" db:"created_at"`
+		UpdatedAt    time.Time `json:"updated_at" db:"updated_at"`
+	}
+
+	// RideChangeSet is the fields that are modifiable in the ride
+	RideChangeSet struct {
+		DriverID     *string   `json:"-"`
+		CarID        *string   `json:"car_id"`
+		Seats        *int      `json:"seats"`
+		StartCity    *string   `json:"start_city"`
+		EndCity      *string   `json:"end_city"`
+		StartLat     *float64  `json:"start_dest_lat"`
+		StartLon     *float64  `json:"start_dest_lon"`
+		EndLat       *float64  `json:"end_dest_lat"`
+		EndLon       *float64  `json:"end_dest_lon"`
+		PricePerSeat *int      `json:"price_per_seat"`
+		Info         *string   `json:"info"`
+		StartDate    time.Time `json:"start_date"`
+	}
+)
+
+// NewRide returns a Ride with the change set fields applied
+func NewRide(r *RideChangeSet) *Ride {
+	ride := Ride{}
+
+	// ignore errors because we will jsut return a zero valued ride
+	ride.ApplyUpdates(r)
+
+	return &ride
+}
+
+// Validate validates a ride
+func (r *Ride) Validate() error {
+	errs := ""
+
+	if r.DriverID == "" {
+		errs += "Driver ID: the ride needs a driver;"
+	}
+
+	if r.DriverID == "" {
+		errs += "Driver ID: the ride needs a driver;"
+	}
+
+	if r.CarID == "" {
+		errs += "Car ID: the ride needs a car;"
+	}
+
+	if r.Seats < 0 {
+		errs += "seats: there needs to be a positive number of seats;"
+	}
+
+	if r.StartCity == "" {
+		errs += "start city: there must be a start city;"
+	}
+
+	if r.EndCity == "" {
+		errs += "end city: there must be a end city;"
+	}
+
+	if r.StartLat < -90 || r.StartLat > 90 {
+		errs += "start lat: latitude is between -90 and 90 degrees;"
+	}
+
+	if r.EndLat < -90 || r.EndLat > 90 {
+		errs += "end lat: latitude is between -90 and 90 degrees;"
+	}
+
+	if r.StartLon < -180 || r.StartLon > 180 {
+		errs += "start lon: longitude is between -180 and 180 degrees;"
+	}
+
+	if r.EndLon < -180 || r.EndLon > 180 {
+		errs += "end lon: longitude is between -180 and 180 degrees;"
+	}
+
+	if r.PricePerSeat < 0 {
+		errs += "price per seat: you must supply a price for the ride;"
+	}
+
+	now := time.Now()
+	if r.StartDate.Before(now) {
+		errs += "StartDate: the ride start date and time must be some time in the future;"
+	}
+
+	if errs != "" {
+		return errors.New(errs)
+	}
+
+	return nil
+}
+
+// String returns a string representation of a ride
+func (r *Ride) String() string {
+	return fmt.Sprintf("<Ride id:%s driver:%s car:%s>", r.ID, r.DriverID, r.CarID)
+}
+
+// ApplyUpdates attempts to update ride and validates the updates
+func (r *Ride) ApplyUpdates(o *RideChangeSet) error {
+	newRide := Ride{}
+
+	if err := mergo.Merge(&newRide, *r, mergo.WithOverride); err != nil {
+		return errors.New("There was a problem applying updates to ride")
+	}
+
```

Comment:
No need for a Merge, just create a copy of the value of existing ride to newRide and update it accordingly. Removes extra dependency.